.github/workflows/test.yaml

```diff
@@ -4,22 +4,22 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.21.x, 1.22.x]
+        go-version: [1.22.x, 1.23.x]
         os: [ubuntu-latest]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Install Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v5
         with:
           go-version: ${{ matrix.go-version }}
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Download test data
-        run: curl -L https://raw.githubusercontent.com/package-url/purl-spec/master/test-suite-data.json -o testdata/test-suite-data.json
+        run: curl -L https://raw.githubusercontent.com/package-url/purl-spec/0dd92f26f8bb11956ffdf5e8acfcee71e8560407/test-suite-data.json -o testdata/test-suite-data.json
```

Comment:
I've taken a look again, and this PR does too many unrelated things in a single commit, IMO. This line is the only one that matches the topic of the commit message AFAIU. It should probably also be annotated with a `# TODO: ...` statement to make the work-around visible without running `git blame`.

All other version updates from this commit should IMO go to another commit in the same PR to clearly separate the intentions.

Reply:
Resolved! I have split all changes into 3 commits.